docs/internal/README.md

```diff
@@ -0,0 +1,130 @@
+# Internal API Docs
+
+## Introduction
+
+CC's internal API is a patchwork of endpoints used by internal components. 
+They evolved over time and were individually designed for specific purposes. 
+We do not recommend using internal API endpoints for anything other than their intended purposes. 
+
+## Endpoints
+
+### POST /internal/v3/staging/:staging_guid/droplet_completed
+**Description:** (Deprecated) Legacy endpoint used to mark droplet staging as complete. Use `POST /internal/v3/staging/:staging_guid/build_completed` instead.
+
+**Intended Consumer:** Diego staging task completion callback
+
+**Auth Mechanism:** MTLS + Basic Auth
+
+### POST /internal/v3/staging/:staging_guid/build_completed
+**Description:** Marks build as completed or failed.
+
+**Intended Consumer:** Diego staging task completion callback
+
+**Auth Mechanism:** MTLS + Basic Auth
+
+### GET /internal/apps/:guid/ssh_access/:index
+**Description:** Check if a user is able to ssh into a process container
+
+**Intended Consumer:** SSH Proxy
+
+**Auth Mechanism:** OAuth
+
+### GET /v2/buildpacks/:guid/download
+**Description:** Download a buildpack file
+
+**Intended Consumer:** ??? (Probably intended for Diego staging task, but it looks like they use direct download from the blobstore now)
+
+**Auth Mechanism:** Basic Auth
+
+### GET /staging/packages/:guid
+**Description:** Download a package if the blobstore is local (NFS server mounted on the CC)
+
+**Intended Consumer:** Diego staging task
+
+**Auth Mechanism:** Basic Auth
+
+### GET /staging/v3/droplets/:guid/download
+**Description:** Download a droplet if the blobstore is local (NFS server mounted on the CC)
+
+**Intended Consumer:** Diego running task or process
+
+**Auth Mechanism:** Basic Auth
+
+### GET /staging/v3/buildpack_cache/:stack/:app_guid/download
+**Description:** Download buildpack cache for a given stack and app combination
+
+**Intended Consumer:** Diego staging task
+
+**Auth Mechanism:** Basic Auth
+
+### GET /internal/v2/droplets/:guid/:droplet_checksum/download
+**Description:** (Deprecated) Permalink for downloading a droplet. Use `GET /internal/v4/droplets/:guid/:droplet_checksum/download` instead.
+
+**Intended Consumer:** Diego running task or process
+
+**Auth Mechanism:** None
```

Comment:
Can you note that this endpoint just redirects to the `/internal/v4` mTLS authenticated one. It doesn't let you actually download the droplet without any auth.

Reply:
I think there is still technically aslo an unauthenticated path: https://github.com/cloudfoundry/cloud_controller_ng/blob/3a93d5fea615fa085b19c2c940c21ad0dfb05948/app/controllers/internal/download_droplets_controller.rb#L24